src/CMakeLists.txt

```diff
@@ -94,8 +112,11 @@ else(no_yasm)
   endif(arch STREQUAL "x86_64")
 endif(no_yasm)
 
-
-set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -ftemplate-depth-1024 -Wnon-virtual-dtor -Wno-invalid-offsetof -Wstrict-null-sentinel -Woverloaded-virtual")
+set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -ftemplate-depth-1024 -Wno-invalid-offsetof")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Woverloaded-virtual")
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
```

Comment:
would be better if we can be consistent regarding to the use of `MATCHES "GNU"` and `STREQUAL GNU`?